### ISSUE TITLE
Magento 2.0.7 SOAP Wsdl List in Json by default

### ISSUE THREAD

**Issue by davidverholen** [magento/magento2#4848]
## Preconditions
1. Install Magento from `2.0` branch.

## Steps to reproduce
1. Install Magento from `2.0` branch.
2. execute `curl http://magento_host/soap/default?wsdl_list=1`
## Expected result
1. Since it is a reference to wsdl resources, I would expect this to be a xml resource by default
## Actual result
1. without the "accept: application/xml" Header, a json Object is returned

**Comment by magento-engcom-team** [comment]
@davidverholen, thank you for your report.
We've created internal ticket(s) MAGETWO-80603 to track progress on the issue.

**Comment by joshuaswarren** [comment]
I'm working on it #SQUASHTOBERFEST #MMNY 

So far I've narrowed it down to Magento/Webapi/Controller/Soap.php - the block starting around line 133, and the fact that the default response type in etc/di.xml is set to:

` <item name="default" xsi:type="array">
                    <item name="type" xsi:type="string">*/*</item>
                    <item name="model" xsi:type="string">Magento\Framework\Webapi\Rest\Response\Renderer\Json</item>
                </item>`

So basically - if no request type is set via a header, the SOAP API is always going to return the WSDL in JSON, even though the SOAP API should be using XML. 

Trying to find a clean way to resolve this without impacting the REST API, but oddly enough, the core SOAP functions are using renderers from the REST API code.

**Comment by joshuaswarren** [comment]
In etc/di.xml, changing:

` <item name="default" xsi:type="array">
                    <item name="type" xsi:type="string">*/*</item>
                    <item name="model" xsi:type="string">Magento\Framework\Webapi\Rest\Response\Renderer\Json</item>
                </item>`

to:

` <item name="default" xsi:type="array">
                    <item name="type" xsi:type="string">*/*</item>
                    <item name="model" xsi:type="string">Magento\Framework\Webapi\Rest\Response\Renderer\Xml</item>
                </item>`

...resolves the issue. I need to see if this breaks the REST API or not. Oddly enough, I don't think the REST API is using this portion of the code. Still testing. 

Update: I've confirmed that, despite the naming and location of the renderers under the REST namespace, in the entire Magento/Webapi module, the only code that uses those renderers is the SOAP API code, so this change appears to be safe.

**Comment by joshuaswarren** [comment]
Per the comment in PR #11501 this change that appeared safe was not, so the pull request I submitted can't be merged in. 

This is definitely a core bug - the SOAP WSDL is being returned by default as JSON-encoded XML, which isn't valid behavior under the SOAP specs. 

However, as I noted in the comments above, the core module that implements the SOAP API is actually using functions from the REST API to render results. There are no similar functions defined in the SOAP API. Someone will need to define those in the SOAP API module, and then change the etc/di.xml lines I mention in my previous comments to reference the new XML-based methods in the SOAP module.

This is beyond the scope of my free time right now, so I'm hoping someone else can pick up work on this. If you do and have questions, let me know and I'll do my best to help!

**Comment by vrann** [comment]
@davidverholen as long as it is not WSDL resource by itself, it is not expected to be in XML format, instead, it uses the default REST serialization mechanism. That mechanism cannot be changed (at least in an easy backward compatible way) for this resource only, so I would suggest using an appropriate header to retrieve the list of WSDL resources in XML.
